atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/OracleTableNameUnmapper.java

```diff
@@ -99,33 +102,49 @@ private Set<String> getBatchOfLongTableNamesFromMappingTable(
 
         // We use the Oracle LOWER function rather than mapping to lower case on the client to ensure that we're
         // using the same locale and conversion.
-        AgnosticResultSet results = conn.selectResultSetUnregisteredQuery(
-                "SELECT table_name FROM " + AtlasDbConstants.ORACLE_NAME_MAPPING_TABLE + " WHERE"
-                        + " LOWER(short_table_name) IN (" + placeHolders + ")",
-                shortTableNames.toArray());
+        Map<String, String> shortTableToLongTableMapping = runIgnoringTableNotFound(
+                () -> {
+                    AgnosticResultSet results = conn.selectResultSetUnregisteredQuery(
+                            "SELECT short_table_name, table_name FROM " + AtlasDbConstants.ORACLE_NAME_MAPPING_TABLE
+                                    + " WHERE" + " LOWER(short_table_name) IN (" + placeHolders + ")",
+                            shortTableNames.toArray());
+
+                    return results.rows().stream()
+                            .collect(Collectors.toMap(
+                                    row -> row.getString("short_table_name"), row -> row.getString("table_name")));
+                },
+                Map::of);
 
-        throwIfResultSetSizeDoesNotMatchExpected(results, shortTableNames);
+        verifyMappingSetSizeNotLargerThanExpected(shortTableToLongTableMapping, shortTableNames);
 
-        return results.rows().stream()
-                .map(result -> result.getString("table_name"))
-                .collect(Collectors.toSet());
+        return shortTableToLongTableMapping;
     }
 
-    private void throwIfResultSetSizeDoesNotMatchExpected(
-            AgnosticResultSet resultSet, Collection<String> shortTableNames) throws TableMappingNotFoundException {
-        if (resultSet.size() < shortTableNames.size()) {
-            throw new TableMappingNotFoundException("Some of the tables " + String.join(", ", shortTableNames)
-                    + " do not have a mapping. This might be because these tables do not exist.");
-        } else if (resultSet.size() > shortTableNames.size()) {
-            throw new SafeIllegalStateException(
-                    "There are more returned long table names than provided short table"
-                            + " names. This likely indicates a bug in AtlasDB",
-                    SafeArg.of("numLongTables", resultSet.size()),
-                    SafeArg.of("numShortTables", shortTableNames.size()));
-        }
+    private void verifyMappingSetSizeNotLargerThanExpected(
+            Map<String, String> shortTableToLongTableMapping, Collection<String> expectedShortTableNames) {
+        Preconditions.checkState(
```

Comment:
This should only really happen if there's something off with our query, or there are multiple different case short table names (but we don't handle case sensitive table names in Atlas anyway). It's here for safety against faulty assumptions